lib/openfeature/sdk/client.rb

```diff
@@ -0,0 +1,70 @@
+# frozen_string_literal: true
+# typed: true
+
+require "sorbet-runtime"
+require "forwardable"
+require "json"
+
+require_relative "./provider/provider"
+require_relative "./evaluation_context"
+require_relative "./metadata"
+require_relative "./hook/hook"
+require_relative "./hook/hook_context"
+require_relative "./evaluation_options"
+require_relative "./resolver/boolean_resolver"
+require_relative "./resolver/number_resolver"
+require_relative "./resolver/object_resolver"
+require_relative "./resolver/string_resolver"
+
+module OpenFeature
+  module SDK
+    # TODO: Write
+    #
+    class Client
+      extend T::Sig
+      extend Forwardable
+
+      class OpenFeatureOptions < T::Struct
+        const :name, T.nilable(String)
+        const :version, T.nilable(String)
+      end
+
+      sig { returns(Metadata) }
+      attr_reader :metadata
+
+      sig { returns(T::Array[Hook]) }
+      attr_accessor :hooks
+
+      def_delegator :@boolean_resolver, :fetch_value, :fetch_boolean_value
+      def_delegator :@boolean_resolver, :fetch_detailed_value, :fetch_boolean_details
+
+      def_delegator :@number_resolver, :fetch_value, :fetch_number_value
+      def_delegator :@number_resolver, :fetch_detailed_value, :fetch_number_details
+
+      def_delegator :@string_resolver, :fetch_value, :fetch_string_value
+      def_delegator :@string_resolver, :fetch_detailed_value, :fetch_string_details
+
+      def_delegator :@object_resolver, :fetch_value, :fetch_object_value
+      def_delegator :@object_resolver, :fetch_detailed_value, :fetch_object_details
```

Comment:
> it might make debugging easier
Implementing `respond_to?` and `respond_to_missing?` should mitigate some of this.

>  I don't think that method_missing does that.

It doesn't own it's own, but can be implemented in a way that defines methods as they go.